src/Symfony/Component/HttpClient/Exception/TransportException.php

```diff
@@ -18,4 +18,13 @@
  */
 final class TransportException extends \RuntimeException implements TransportExceptionInterface
 {
+    public static function connectionTimeoutReached(string $url, float $seconds): self
```

Comment:
we have very limited use of exception factories in the codebase, they make stack traces more complex